### ISSUE TITLE
train on multi gpus

### ISSUE THREAD

**Issue by cherryxiongyw** [multimodallearning/pytorch-mask-rcnn#10]
I want to train this model on multi gpus, so I add codes in coco.py likes this：
`model = torch.nn.DataParallel(model,device_ids=[0,1])    
  model = model.cuda()`
but when I run, it still only runs on gpu0, so what happened? How can I do?
Thanks!

**Comment by lasseha** [comment]
As this is more of a pytorch related question i would refer to the pytorch tutorials and docs. If there are problems with training on multiple gpus and this particular mask rcnn implementation feel free to ask again.

**Comment by cherryxiongyw** [comment]
@lasseha 
Thanks for your help! Today I find another problem for pytorch, when I train on one gpu with batchsize=4, it is not faster then that where batchsize=1. For example, when batchsize=1 it will cost 1s, it will cost about 4s when batchsize=4. So I can't find it will be faster when batchsize is set larger. I don't know why?

**Comment by rcampos92** [comment]
I tested your code and I started to have good results using your previous pre-trained model, I would like to know where could be the best place to put this command to give support for multiple GPUs. The created class for the network does not have a specific place to call the model to apply easily this function. Can you include this function somewhere?

model_multiple_gpu = torch.nn.DataParallel(model, device_ids=range(torch.cuda.device_count())) 

Thank you in advance.

**Comment by lasseha** [comment]
@hli2020 implemented a multi-gpu version based on this repository. Follow issue #19.
